### PR TITLE
Refactor pipelines - Spring Grove

### PR DESCRIPTION
**Pull request by scbedd**
Related to Azure/azure-sdk-tools#7941 

This PR will:

- [x] Update CI job, regression job, and release job definitions to be in the `azure-sdk` project under the `tools` folder
- [x] Standardize oav build and release similar to the rest of the azure-sdk project standards
- [x] Enable `1es-templates` for the repository
- [x] Centralize regression jobs into this job
- [x] Get CI stable
- [x] public/private/release build conditions tested and working
- [x] ~~Test release (blocked on this until `microsoft1es` is added to the `oav` project on npmjs.org)~~ They are just delaying. Merging this.
- [x] Update regression snapshots due to updated file locations (new agents mean new root)
- [x] Stabilize SDL sources error [active question on support channel](https://teams.microsoft.com/l/message/19:6d24d44587aa4d868bd4e96fe0eead31@thread.tacv2/1717098256390?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=886b1d22-e648-4492-af5f-6fbe81e73d4a&parentMessageId=1717098256390&teamName=1ES%20Pipeline%20Templates%20(1ES%20PT)&channelName=Support%20-%20General&createdTime=1717098256390)

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

File: ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Change Log - oav
 
+## 06/03/2024 3.3.7
```
Comment (mikeharder) [inline ChangeLog.md]: I didn't publish 3.3.6, so unless you did, I think we can re-use 3.3.6 for the combined changes.

File: ci.yml
Comment (mikeharder) [inline ci.yml:1]: I would move straight to GH Actions, unless there's a reason this should be a DevOps pipeline.

File: ChangeLog.md (hunk 1 above)
Comment (scbedd) [reply ChangeLog.md]: Absolutely right.

File: ci.yml
Comment (scbedd) [reply ci.yml:1]: Production pipeline need publishing to npm. No actions for now 👍

File: eng/templates/stages/oav.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+              os: linux
+
+            steps:
+              - checkout: self
+                submodules: true
+
+              - task: NodeTool@0
+                inputs:
+                  versionSpec: "20.x"
+                displayName: "Use Node 20"
```
Comment (mikeharder) [inline eng/templates/stages/oav.yml:26]: It would be good to test on a sparse matrix of Linux/Windows and Node18/Node20, like I do here:

https://github.com/Azure/azure-rest-api-specs/blob/bf9738223ed8c1c25485980b74cfb8ef4158b847/.github/workflows/_reusable-eng-tools-test.yaml#L15

File: package.json [hunk 3]
```diff
@@ -13,6 +13,8 @@
     "@autorest/schemas": "^1.3.4",
     "@azure-tools/openapi-tools-common": "^1.2.2",
     "@azure/core-http": "^3.0.1",
+    "@azure/core-util": "~1.7.0",
```
Comment (mikeharder) [inline package.json:16]: Why are these needed?  Old TypeScript version?

**Review by mikeharder — APPROVED** [review]

File: eng/templates/stages/oav.yml (hunk 2 above)
Comment (scbedd) [reply eng/templates/stages/oav.yml:26]: I agree. Tests crash on windows currently, and I don't want to deal with that at this time,

File: package.json (hunk 3 above)
Comment (scbedd) [reply package.json:16]: Exactly why. When I bump TS version these will go away.

**Comment by scbedd** [comment]
Bypassing these restrictions, new builds will work after this PR merges to `develop`.
